### PR TITLE
Discovery community content fixes

### PR DESCRIPTION
**Pull request by eldiegod**
fixes: #1616, #1637, #1636 and  some other details found along the way

### CODE REVIEW

File: src/components/_video/VideoCategoryCard/FeaturedVideoCategoryCard.tsx [hunk 1]
```diff
@@ -48,21 +50,14 @@ export const FeaturedVideoCategoryCard: React.FC<FeaturedVideoCategoryCardProps>
...
           <PlayerContainer>
-            {
-              <BackgroundVideoPlayer
-                src={isLoading ? undefined : videoUrl}
-                loop
-                muted
-                playing={variant === 'default' ? isVideoHovering : true}
-              />
-            }
+            {<BackgroundVideoPlayer src={isLoading ? undefined : videoUrl} loop muted playing={isPlaying} />}
```
Comment (drillprop) [inline src/components/_video/VideoCategoryCard/FeaturedVideoCategoryCard.tsx]:
```suggestion
            <BackgroundVideoPlayer src={isLoading ? undefined : videoUrl} loop muted playing={isPlaying} />
```
We don't need curly braces here

File: src/views/viewer/DiscoverView/DiscoverView.tsx [hunk 2]
```diff
@@ -25,22 +26,28 @@ export const DiscoverView: React.FC = () => {
...
+            categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
+            categoriesFeaturedVideos?.[category.id]?.[0] ??
+            undefined
+          return video === undefined
+            ? undefined
+            : {
+                videoTitle: video?.title ?? '',
+                videoUrl: video?.videoCutUrl ?? '',
+                ...mappedVideoCategories?.find((cat) => cat.id === category.id),
+              }
```
Comment (drillprop) [inline src/views/viewer/DiscoverView/DiscoverView.tsx]:
```suggestion
          if (video) {
            return {
              videoTitle: video?.title ?? '',
              videoUrl: video?.videoCutUrl ?? '',
              ...mappedVideoCategories?.find((cat) => cat.id === category.id),
            }
          }
```

File: src/views/viewer/DiscoverView/DiscoverView.tsx [hunk 3]
```diff
@@ -25,22 +26,28 @@ export const DiscoverView: React.FC = () => {
...
-    })
-    .filter((cat) => !!cat) ?? [null, null, null]
+  const featuredVideoCategoryCardsData = React.useMemo(() => {
+    const _featuredVideoCategoryCardsData = sampleSize(
+      categories
+        ?.map((category) => {
+          const video =
+            categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
+            categoriesFeaturedVideos?.[category.id]?.[0] ??
+            undefined
```
Comment (drillprop) [inline src/views/viewer/DiscoverView/DiscoverView.tsx]:
```suggestion
          const video =
            categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
            categoriesFeaturedVideos?.[category.id]?.[0]
```
Could be wrong, but I don't think we need this `?? undefined` here.

File: src/api/featured/categoriesFeaturedVideos.ts [hunk 4]
```diff
@@ -27,11 +28,15 @@ export const useCategoriesFeaturedVideos = (): CategoriesFeaturedVideos | null =
...
+    { limit: allVideosIds?.length, where: { id_in: allVideosIds } },
+    { skip: !allVideosIds || !allVideosIds.length }
+  )
   const videosLookup = createLookup(videos || [])
-
   const categoriesLookup = rawData?.reduce((acc, cur) => {
-    acc[cur.categoryId] = cur.videos.map((v) => ({ ...videosLookup[v.videoId], videoCutUrl: v.videoCutUrl }))
+    acc[cur.categoryId] = cur.videos.map((v) => {
+      return { ...videosLookup[v.videoId], videoCutUrl: v.videoCutUrl }
+    })
```
Comment (drillprop) [inline src/api/featured/categoriesFeaturedVideos.ts]:
```suggestion
    acc[cur.categoryId] = cur.videos.map((v) => ({ ...videosLookup[v.videoId], videoCutUrl: v.videoCutUrl }))
```

**Review by drillprop — COMMENTED** [review]
I see also two warnings in the console: 
image
image
This is probably not related to your changes, I think it was there before. Could you fix them as part of this PR as well?

File: src/views/viewer/DiscoverView/DiscoverView.tsx (hunk 3 above)
Comment (drillprop) [reply src/views/viewer/DiscoverView/DiscoverView.tsx]: Actually, we can just do: 
```tsx
const video = categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl)
```
Right🤔 ?

Comment (eldiegod) [reply src/views/viewer/DiscoverView/DiscoverView.tsx]: yes this works

File: src/views/viewer/DiscoverView/DiscoverView.tsx (hunk 2 above)
Comment (eldiegod) [reply src/views/viewer/DiscoverView/DiscoverView.tsx]: I rather keep it explicit that we are possibly returning undefined here

Comment (drillprop) [reply src/views/viewer/DiscoverView/DiscoverView.tsx]: Hmm, I don't agree with this. I mean, it's quite clear for me, that if we don't return from `Array.map` we will just return `undefined`. Your current code gives me the impression that something else going on here, and I need to decode this every time I'm looking at these lines 😅 . 
Of course, this is only my opinion. Not sure if everyone on the team agrees with me. Let's leave that unresolved for now, maybe?

Comment (kdembler) [reply src/views/viewer/DiscoverView/DiscoverView.tsx]: I think we actually shouldn't rely on `undefined` that much, `null` seems to be a better fit here as it explicitly tells you the value is missing. How about this, should be both readable and explicit:
```tsx
const video =
  categoriesFeaturedVideos?.[category.id]?.find((video) => !!video.videoCutUrl) ??
  categoriesFeaturedVideos?.[category.id]?.[0]

if (!video) return null

return {
  // ...
}
```

**Review by kdembler — CHANGES_REQUESTED** [review]
Found some more issues:
1. The thumbnails in Video Hero don't have a proper loading state:

https://user-images.githubusercontent.com/12646744/143020714-93082044-6008-4445-a2ed-6900b51d51fb.mov

2. We're missing bottom space in `Discover` view
3. How do we pick featured categories? They seem to be different on every load, also if I go to specific category and navigate back, the list will change. We should have those at least semi-consistent. I'm not sure what the correct long-term solution is, for the time being maybe we just hardcode IDs or just show the first 3 categories? Open to suggestions on this one

**Comment by kdembler** [comment]
I think we just improve the loading of thumbnails inside video hero. You've added placeholders for images but even if those are loading, we should still show the progress bar indicating which video is currently playing.

**Comment by kdembler** [comment]
The video hero thumbnails still don't work properly when loading:

https://user-images.githubusercontent.com/12646744/143212152-40be34cc-bef3-40e2-8f56-f7bd6c162ebe.mov

The progress bar should be spaced from the skeleton and also seems like the progress bar has a different color when the thumbnail is loading? Maybe an overlap?

**Review by kdembler — APPROVED** [review]

**Pull request merged**